lib/terraforming/resource/ec2.rb

```diff
@@ -96,6 +96,10 @@ def in_vpc?(instance)
           (instance.subnet_id && instance.subnet_id != "" && instance.security_groups.empty?)
       end
 
+      def associate_public_ip?(instance)
+        (instance.public_ip_address.to_s.empty?) ? "false" : "true"
+      end
```

Comment:
This method should return boolean value:

```ruby
def associate_public_ip_address?(instance)
  !instance.public_ip_address.to_s.empty?
end
```

and both `tf` and `tfstate` should call this method.

tf: 

```
associate_public_ip_address = <%= associate_public_ip_address?(instance) %>
```

tfstate:

```
"associate_public_ip_address?(instance).to_s
```